graveler/listing_iterator.go

```diff
@@ -0,0 +1,124 @@
+package graveler
+
+import "bytes"
+
+// ListingIter implements a listing iterator using a ValueIterator
+// assumes all values in valueIterator start with prefix
+type ListingIter struct {
+	valueIterator ValueIterator
+	delimiter     Key
+	nextDelimiter Key
+	prefix        Key
+	current       *Listing
+	nextFunc      func(l *ListingIter) bool
```

Comment:
```suggestion
	nextFunc      func() bool
```

turn the functions into methods